libsolidity/ast/Types.cpp

```diff
@@ -1267,7 +1267,11 @@ shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
 	if (v > u256(-1))
 		return shared_ptr<FixedPointType const>();
 
-	unsigned totalBits = max(bytesRequired(v), 1u) * 8;
+	boost::optional<unsigned> requiredBytes = bytesRequired(v, 4096);
```

Comment:
I don't think this is the proper fix. Numbers that are too large should be caught by the `if (v > u256(-1))` check above.

Reply:
Compare with ``integerType()`` - five lines above in the ``if (negative)`` case the sign should be flipped, but isn't.

Reply:
I'm pretty sure now that the problem is that in line 1265, we end up with `v==0` and `negativ == true`. In this case, I think the fix would be to change `if (negative)` to `if (negative && v != 0)`, but please provide specific example values first, I would like to know how exactly this situation can happen.

Reply:
The scenario can be reproduced with `fixed n = 12222222222202 ** -22 * -21111111111121;`.

Reply:
ah, forgot to mention @chriseth: adding `&& v != 0` in the `if (negative)` fixes it and sets `totalBits` to `8`, which feels right.